foundation-ui/mount/MountPoint.js

```diff
@@ -0,0 +1,109 @@
+import React, {PropTypes} from 'react';
+
+import MountService from './MountService';
+import {CHANGE} from './MountEvent';
+import ReactUtil from '../utils/ReactUtil';
+
+const METHODS_TO_BIND = [
+  'handleMountServiceChange'
+];
+
+/**
+ * A component to use as a mount point to incorporate views/components provided
+ * by packages. It renders with a/components registered to the `MountService`
+ * with a matching `type`.
+ * It provides a property to configure the wrapping component (`wrapper`) as
+ * well as a property to `limit` the number of rendered elements.
+ *
+ * @example
+ *
+ * <MountPoint type="widget" limit={10} wrapper={MyWidgetsWrapper}>
+ *     <span className="help-text"></span>
+ * </MountPoint>
+ *
+ */
+class MountPoint extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    // Get components and init state
+    this.state = {
+      components: MountService.findComponentsWithType(this.props.type)
+    };
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentWillMount() {
+
+    MountService.addListener(CHANGE, this.handleMountServiceChange);
+  }
+
+  componentWillReceiveProps({type}) {
+    if (this.props.type === type) {
+      return;
+    }
+
+    this.setState({components: MountService.findComponentsWithType(type)});
+  }
+
+  componentWillUnmount() {
+    MountService.removeListener(CHANGE, this.handleMountServiceChange);
+  }
+
+  handleMountServiceChange(type) {
+    if (this.props.type !== type) {
+      return;
+    }
+
+    this.setState({components: MountService.findComponentsWithType(type)});
+  }
+
+  render() {
+    const {alwaysWrap, children, limit, wrapper} = this.props;
+    const {components} = this.state;
+
+    // Filter consumed props to only pass on a "clean" set of props
+    const filteredProps = Object.keys(this.props).reduce((props, key) => {
+      if (!Object.prototype.hasOwnProperty.call(MountPoint.propTypes, key)) {
```

Comment:
They're included but I'm going to introduce an additional list so that we don't fail once we add an additional optimization step which removes them.